packages/protocol/test-sol/unit/governance/network/Governance.t.sol

```diff
@@ -3109,112 +3267,311 @@ contract GovernanceTest_prepareHotfix is GovernanceTest {
-    approveAndPrepareHotfix();
+  function test_Emits_HotfixPreparedEvent_whenHotfixApproved() public {
+    vm.prank(accOwner);
+    governance.setHotfixExecutionTimeWindow(DAY);
+    vm.prank(accCouncil);
+    governance.approveHotfix(HOTFIX_HASH);
+    vm.prank(accApprover);
+    governance.approveHotfix(HOTFIX_HASH);
+
     vm.expectEmit(true, true, true, true);
-    emit HotfixExecuted(hotfixHash);
-    executeHotfixTx();
+    emit HotfixPrepared(HOTFIX_HASH, block.timestamp + DAY);
+    governance.prepareHotfix(HOTFIX_HASH);
+  }
+
+  function test_Reverts_IfHotfixExecutionTimeWindowNotSet() public {
+    vm.prank(accCouncil);
+    governance.approveHotfix(HOTFIX_HASH);
+    vm.prank(accApprover);
+    governance.approveHotfix(HOTFIX_HASH);
+
+    vm.expectRevert("Hotfix execution time window not set");
+    governance.prepareHotfix(HOTFIX_HASH);
+  }
+  function test_Reverts_IfHotfixIsNotApproved() public {
+    vm.prank(accOwner);
+    governance.setHotfixExecutionTimeWindow(DAY);
+
+    vm.expectRevert("Hotfix not approved by approvers.");
+    governance.prepareHotfix(HOTFIX_HASH);
+
+    vm.prank(accApprover);
+    governance.approveHotfix(HOTFIX_HASH);
+    vm.expectRevert("Hotfix not approved by security council.");
+    governance.prepareHotfix(HOTFIX_HASH);
+  }
+
+  function test_Reverts_IfPreparedTwiceWithinExecutionTimeLimit() public {
+    vm.prank(accOwner);
+    governance.setHotfixExecutionTimeWindow(DAY);
+    vm.prank(accCouncil);
+    governance.approveHotfix(HOTFIX_HASH);
+    vm.prank(accApprover);
+    governance.approveHotfix(HOTFIX_HASH);
+
+    governance.prepareHotfix(HOTFIX_HASH);
+    vm.expectRevert("Hotfix already prepared for this timeframe.");
+    governance.prepareHotfix(HOTFIX_HASH);
+  }
+}
+
+contract GovernanceTest_resetHotfix is GovernanceTest {
+  bytes32 constant HOTFIX_HASH = bytes32(uint256(0x123456789));
+  bytes32 constant SALT = 0x657ed9d64e84fa3d1af43b3a307db22aba2d90a158015df1c588c02e24ca08f0;
+  bytes32 hotfixHash;
+  event HotfixRecordReset(bytes32 indexed hash);
+
+  function setUp() public {
+    super.setUp();
+    _whenL2();
+    vm.prank(accOwner);
+    governance.setSecurityCouncil(accCouncil);
+
+    hotfixHash = governance.getHotfixHash(
+      okProp.values,
+      okProp.destinations,
+      okProp.data,
+      okProp.dataLengths,
+      SALT
+    );
+  }
+  function test_ShouldResetHotfixRecordWhenExecutionTimeLimitHasPassed() public {
+    vm.prank(accOwner);
+    governance.setHotfixExecutionTimeWindow(DAY);
+
+    vm.prank(accCouncil);
+    governance.approveHotfix(HOTFIX_HASH);
+    vm.prank(accApprover);
+    governance.approveHotfix(HOTFIX_HASH);
+
+    governance.prepareHotfix(HOTFIX_HASH);
+    timeTravel(DAY + 1);
+    governance.resetHotFixRecord(HOTFIX_HASH);
+
+    (bool approved, bool councilApproved, , uint256 _preparedTimeLimit) = governance
+      .getL2HotfixRecord(HOTFIX_HASH);
+    assertFalse(approved);
```

Comment:
this test could have. false postive i believe if the approveHotfix fails it will never be true. could add an intermediate assertion